lib/APMTracer.js

```diff
@@ -156,46 +159,51 @@ class APMTracer {
      *         be found in `carrier`
      */
     extract(format, carrier) {
-        if (format === Constants.FORMAT_HTTP_HEADERS ||
-            format === Constants.FORMAT_TEXT_MAP) {
-            return APMTracer._extractHttpAndTextMap(carrier);
+        let spanContext;
+
+        switch (format) {
+        case Constants.FORMAT_HTTP_HEADERS:
+        case Constants.FORMAT_TEXT_MAP:
+            spanContext = APMTracer._extractHttpAndTextMap(carrier);
+            break;
+        default:
+            console.log('Extract unknown format');
+            spanContext = new APMSpanContext();
         }
 
-        console.log('Unsupported format');
-        return new APMSpanContext();
+        spanContext.setInitialConsumer();
+        return spanContext;
     }
 
     static _extractHttpAndTextMap(carrier) {
         let spanId;
         let traceId;
         Object.keys(carrier).forEach((objectKey) => {
-            switch (objectKey.toLowerCase()) {
+            switch (objectKey.toUpperCase()) {
             case CARRIER_SPAN_ID:
                 spanId = carrier[objectKey];
                 break;
             case CARRIER_TRACE_ID:
                 traceId = carrier[objectKey];
                 break;
-            case CARRIER_SAMPLED:
-                break;
-            // ignoring
             default:
                 // error, but ignoring
             }
         });
 
+        const spanContext = new APMSpanContext(null, null, null);
+
         if (spanId && traceId) {
-            return new APMSpanContext(spanId, traceId);
+            spanContext._spanId = spanId;
+            spanContext._traceId = traceId;
         }
 
-        // span context not found
-        return new APMSpanContext();
+        return spanContext;
     }
 }
 
 module.exports = {
     APMTracer,
     CARRIER_TRACE_ID,
     CARRIER_SPAN_ID,
```

Comment:
May be better to name this just as ID - to avoid potential confusion with the span id that is propagated with zipkin.